x-pack/plugins/apm/server/lib/helpers/get_kuery_fields.test.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { getKueryFields } from './get_kuery_fields';
+import { fromKueryExpression } from '@kbn/es-query';
+
+describe('get kuery fields', () => {
```

Comment:
can you add a test here that uses groups, e.g. `(service.name:opbeans-* OR service.name:kibana) and (service.environment:production)`, and one for queries without field names, e.g. just `opbeans`)?

Reply:
Good point. Also, how do we handle syntax like `service.name: "foo` (missing end quote)?
If it throws, we should probably handle it so we don't crash the telemetry collection.

Reply:
both great points. 
I address the first one (here)[https://github.com/elastic/kibana/pull/132789/commits/1a21592994bf0f6a4b95bbe195ccfb71b75545f9]

We don't allow the user to create a service group with syntax like this `service.name: "foo` BUT he can edit the query without any validation so I need to handle this case as well